Paulvitalis200/Store-Manager-API-V2
# #161618211 Admin wrapper

#### What does this PR do?
This PR creates an admin wrapper function

#### Description of Task to be completed?
Create an admin wrapper function that will be called on the methods which only the admin should have access to.

#### How should this be manually tested?
Clone this repository using git clone `https://github.com/Paulstar200/Store-Manager-API-V2.git`
Create a database on your local machine called `store_manager_api`
Switch to the branch `ft-admin-wrapper-161618211` by doing `git checkout ft-admin-wrapper-161618211`
On your terminal, install a virtual environment using `pip install virtualenv`
Create a virtual environment using `python -m virtualenv env`
Activate the virtual environment using `env\scripts\activate` for windows
For Mac use `env/bin/activate`
On your terminal install the dependencies using `pip install -r requirements.txt`
Do `set FLASK_APP=run.py`
Run the app in your terminal using `python run.py` or `flask run`
Copy the localhost link and test it using Postman.
Login to the app so that you get an access token.
Add the bearer token to your header using the format `Bearer <access token>`
Try and post a product without logging in as the admin
Post a product first using Post and adding to the url `/api/v2/products/`
{
 "name": "Shoes",
"quantity": 34,
"price": 200
}
There should be printed an error stating that only admin has access to the following method


#### Any background context you want to provide?
Ensure the setup of the database is complete before you run the application

#### What are the relevant pivotal tracker stories?
#161618211

#### Screenshots (if appropriate)
None

#### Questions:
Feel free to ask any questions
